### DISCUSSION TITLE
Version 5.2.0-alpha.2

### DISCUSSION THREAD

**Discussion by coolsoftwaretyler** [mobxjs/mobx-state-tree#2076] [General] [answered]
A follow up to https://github.com/mobxjs/mobx-state-tree/releases/tag/v5.2.0-alpha.1 and second release candidate for `5.2.0`.

##  Breaking changes

* No breaking changes

## Features

* Add flag to determine if an action is a flow based action by @Slooowpoke in https://github.com/mobxjs/mobx-state-tree/pull/1856
* give complex types debug names by @BrianHung in https://github.com/mobxjs/mobx-state-tree/pull/2049
* Add lazy type for lazy importing of models by @clgeoio in https://github.com/mobxjs/mobx-state-tree/pull/1722
* Add two new primitives for MST by @chakrihacker in https://github.com/mobxjs/mobx-state-tree/pull/2052

## Fixes

* fix: throw error when creating a model with faulty name and valid pro… by @nithinssabu in https://github.com/mobxjs/mobx-state-tree/pull/2048
* fix splitCache, extra nodes were being removed by @scytacki in https://github.com/mobxjs/mobx-state-tree/pull/2054
* call lifecycle hooks after children are created by @scytacki in https://github.com/mobxjs/mobx-state-tree/pull/1952
* Allow readonly string arrays in types.enumeration by @jamonholmgren in https://github.com/mobxjs/mobx-state-tree/pull/2059
* fix: add checks while creating model by @chakrihacker in https://github.com/mobxjs/mobx-state-tree/pull/2057
* fix: revert parent context features by @coolsoftwaretyler in https://github.com/mobxjs/mobx-state-tree/pull/2074

## Docs

* docs: update references to example repos that will move external by @coolsoftwaretyler in https://github.com/mobxjs/mobx-state-tree/pull/2042
* docs: remove extra reference of packages/mst-example-bookshop by @JuanJo4 in https://github.com/mobxjs/mobx-state-tree/pull/2047
* Docs: fix some grammar and typos by @kylemeenehan in https://github.com/mobxjs/mobx-state-tree/pull/2070

## Tests

* Test/add more test coverage for types model by @coolsoftwaretyler in https://github.com/mobxjs/mobx-state-tree/pull/2044
* Test/add property parsing tests by @coolsoftwaretyler in https://github.com/mobxjs/mobx-state-tree/pull/2051
* Add a test for debug name by @BrianHung in https://github.com/mobxjs/mobx-state-tree/pull/2069

## Development/community

* chore: update GitHub issue templates to focus on bugs by @coolsoftwaretyler in https://github.com/mobxjs/mobx-state-tree/pull/2017
* chore: remove examples from repository by @coolsoftwaretyler in https://github.com/mobxjs/mobx-state-tree/pull/2043
* chore: add pull request template by @coolsoftwaretyler in https://github.com/mobxjs/mobx-state-tree/pull/2066

## New Contributors
* @coolsoftwaretyler made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/2017
* @Slooowpoke made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/1856
* @JuanJo4 made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/2047
* @nithinssabu made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/2048
* @BrianHung made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/2049
* @scytacki made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/2054
* @clgeoio made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/1722
* @chakrihacker made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/2052
* @kylemeenehan made their first contribution in https://github.com/mobxjs/mobx-state-tree/pull/2070

**Full Changelog**: https://github.com/mobxjs/mobx-state-tree/compare/v5.1.8...v5.2.0-alpha.2

<hr /><em>This discussion was created from the release <a href='https://github.com/mobxjs/mobx-state-tree/releases/tag/v5.2.0-alpha.2'>Version 5.2.0-alpha.2</a>.</em>

**Comment by coolsoftwaretyler** [comment]
This release candidate is live on npm!

https://www.npmjs.com/package/mobx-state-tree?activeTab=versions

**Comment by coolsoftwaretyler** [comment]
I've been running this in production for about a week to good success. Planning to mark as `latest` sometime friday.

**Reply by chakrihacker** [reply]
Let's do on Monday

**Reply by coolsoftwaretyler** [reply]
@chakrihacker - I won't have a lot of time to do that on Monday. I would prefer to promote on Friday. Is there a reason you want me to wait?

**Reply by chakrihacker** [reply]
If some one faces issue on the weekends it would ruin their day, so

**Reply by coolsoftwaretyler** [reply, marked as answer]
Fair enough. I won't be available on Monday but we can go for Tuesday. I'll make a Twitter post about it as well and we can get another week of usage out in the wild on the release candidate.

Thanks!

**Comment by coolsoftwaretyler** [comment]
Closing this out since it's been open for a while
